https://issues.redhat.com/browse/ACM-13654 (#6842)
open-cluster-management/rhacm-docs: observability/observability_arch.adoc

The `multiclusterhub-operator` enables the `multicluster-observability-operator` pod by default. You must configure the `multicluster-observability-operator` pod.

- <<obs-open-components,Observability open source components>>
- <<arch-diagram,Observability architecture diagram>>
- <<persistent-stores-observability,Persistent stores used in the observability service>>

[#obs-open-components]
== Observability open source components

> Observability service uses exisiting and widely-adopted observabiliity tools from the open source community. View the following descriptions of the tools that are apart of the product observability service:

Author: ```suggestion
Observability service uses open source observability tools from community. View the following descriptions of the tools that are apart of the product observability service:
```

> - *Thanos:* A query tool you can use to perform global querying across multiple Prometheus instances, and for long-term storage of Prometheus data and persist it in any S3 compatible storage. You can also compose a highly-available and scalable metrics system.

Reviewer: ```suggestion
- *Thanos:* A toolkit of components you can use to perform global querying across multiple Prometheus instances, and for long-term storage of Prometheus data by persisting it in any S3 compatible storage. You can compose a highly-available and scalable metrics system.
```

Reviewer 2: I think this is a really long sentence. If we are to keep it, for translation and accessibility purposes we need the following:

A toolkit of components **_that_** you can use to perform global querying across multiple Prometheus instances, 

-----
"and for long-term storage of Prometheus data by persisting it in any S3 compatible storage."

This line reads very confusing. I think this needs to be a stand-alone thought.

Reviewer 2: @dockerymick I had to re-peer review with this change, FYI. I had approved but we need to work on line 16.

Reviewer: Yes, sorry, just wanted the terminology to be correct. I'm fine splitting sentences!

Reviewer 2: _For long-term storage of Prometheus data, persist it in any S3 compatible storage._

Reviewer 2: If that works @saswatamcode, that is my only recommended change. We try not to have longer sentences in the doc because a non-sighted user with a screen reader would have a lot to digest.

Reviewer 2: no need to apologize :-)

> - Thanos: Thanos is a tool you can use to perform global querying across multiple Prometheus instances. You can also use it for long-term storage of Prometheus data and persist it in any S3 compatible storage. Thanos is a toolkit of components that allow you to compose a highly-available and scalable metrics system.

Reviewer 2: This is a definition list, please see the manual for how to build that. 

And we typically don't repeat the word in the definition, you could break these out so they are easier to read, and reduce some of the length of text.

For all of these, the descriptions can be more concise without as many "it" and repeated words like "You can":

**Thanos:** A monitoring and alerting tool that you can use to collect metrics from your application

Store metrics as time series data. Store all scrapped samples locally. Run rules to aggregate (_misspelled?_) or record new time series from existing data, or generate alerts. (_is it OR and not AND? generate alerts_)

something to consider: https://brainly.com/question/46524136 (given that time and series are joined to describe "data". If not, it could be read as time series-data.

Reviewer 2: https://www.ibm.com/docs/en/ibm-style?topic=format-lists#definition-lists

- Prometheus: Prometheus is a monitoring and alerting tool that you can use to collect metrics from your application and store these metrics as time series data. You can store all scraped samples locally and run rules to aggregrate or record new time series from existing data, or generate alerts.

- Alertmanager: Alertmanager is used to manage alerts from Prometheus. You can use it to receive alerts from multiple Prometheii. You can deduplicate, group, and route alerts to your integrations such as email, Slack, and PagerDuty. You can also configure Alertmanager to silence and inhibit specific alerts.

> 

Reviewer 2: same for all these, could be reduced and build as a definition list with a paragraph indented below each one.

The `multiclusterhub-operator` enables the `multicluster-observability-operator` pod by default. You must configure the `multicluster-observability-operator` pod.

> When the Observability service is enabled, the hub cluster is always configured to collect and send metrics to the configured Thanos instance, regardless of whether hub self-management is enabled or not. When the hub cluster is self-managed, the `disableHubSelfManagement` parameter is set to `false`, which is the default setting. Metrics and alerts for the hub cluster appear in the `local-cluster` namespace. The `local-cluster` only appears in the cluster list drop-down menu if hub self-management is enabled. You can query the `local-cluster` metrics in the Grafana explorer. View the following topics to learn more:

Author: ```suggestion
```

The `multiclusterhub-operator` enables the `multicluster-observability-operator` pod by default. You must configure the `multicluster-observability-operator` pod.

> When the Observability service is enabled, the hub cluster is always configured to collect and send metrics to the configured Thanos instance, regardless of whether hub self-management is enabled or not. When the hub cluster is self-managed, the `disableHubSelfManagement` parameter is set to `false`, which is the default setting. Metrics and alerts for the hub cluster appear in the `local-cluster` namespace. The `local-cluster` only appears in the cluster list drop-down menu if hub self-management is enabled. You can query the `local-cluster` metrics in the Grafana explorer. View the following topics to learn more:

Reviewer: Not entirely sure what this section is talking about

> When the Observability service is enabled, the hub cluster is always configured to collect and send metrics to the configured Thanos instance, regardless of whether hub self-management is enabled or not. When the hub cluster is self-managed, the `disableHubSelfManagement` parameter is set to `false`, which is the default setting. Metrics and alerts for the hub cluster appear in the `local-cluster` namespace. The `local-cluster` only appears in the cluster list drop-down menu if hub self-management is enabled. You can query the `local-cluster` metrics in the Grafana explorer. View the following topics to learn more:

Author: This can go into the configuration file (observe_environments.adoc)

- <<obs-open-components,Observability open source components>>
- <<arch-diagram,Observability architecture diagram>>
- <<persistent-stores-observability,Persistent stores used in the observability service>>

[#obs-open-components]
== Observability open source components

Observability service uses exisiting and widely-adopted observabiliity tools from the open source community. View the following descriptions of the tools that are apart of the product observability service:

//what components do we want to mention. based on the table later in this file, it seems liek we might want to mention: compactor, rule, receive, shard 

> - Thanos: Thanos is a tool you can use to perform global querying across multiple Prometheus instances. You can also use it to store data from Prometheus disl to S3 compatible storage. The following components can be deployed independently and are included within Thanos: metric sources, stores, and queriers.

Reviewer: ```suggestion
- Thanos: Thanos is a tool you can use to perform global querying across multiple Prometheus instances. You can also use it for long-term storage of Prometheus data and persist it in any S3 compatible storage. Thanos is a toolkit of components that allow you to compose a highly-available and scalable metrics system.
```

//I read that there are optional components within Prometheus, which ones do we want to mention here? | MJ | 09/10

> - Prometheus: Prometheus is a monitoring and alerting toolkit that you can use to collect and store metrics as time series data. You can store all scraped samples locally and run rules to aggregrate or recored new time series from existing data, or generate alerts.

Reviewer: ```suggestion
- Prometheus: Prometheus is a monitoring and alerting tool that you can use to collect metrics from your application and store these metrics as time series data. You can store all scraped samples locally and run rules to aggregrate or record new time series from existing data, or generate alerts.
```

> //what components do we want to mention. based on the table later in this file, it seems liek we might want to mention: compactor, rule, receive, shard 
> - Thanos: Thanos is a tool you can use to perform global querying across multiple Prometheus instances. You can also use it for long-term storage of Prometheus data and persist it in any S3 compatible storage. Thanos is a toolkit of components that allow you to compose a highly-available and scalable metrics system.
> 
> //I read that there are optional components within Prometheus, which ones do we want to mention here? | MJ | 09/10
> - Prometheus: Prometheus is a monitoring and alerting tool that you can use to collect metrics from your application and store these metrics as time series data. You can store all scraped samples locally and run rules to aggregrate or record new time series from existing data, or generate alerts.

Author: ```suggestion
- Thanos: Thanos is a tool you can use to perform global querying across multiple Prometheus instances. You can also use it for long-term storage of Prometheus data and persist it in any S3 compatible storage. Thanos is a toolkit of components that allow you to compose a highly-available and scalable metrics system.

- Prometheus: Prometheus is a monitoring and alerting tool that you can use to collect metrics from your application and store these metrics as time series data. You can store all scraped samples locally and run rules to aggregrate or record new time series from existing data, or generate alerts.
```

> - Alertmanager: Alertmanager is a component of Prometheus that you can use to receive alerts from client applications. You can deduplicate, group, and route alerts to receiver integrations such as email, Slack, and PagerDuty. You can also configure Alertmanager to silence and inhibit specific alerts.

Reviewer: ```suggestion
- Alertmanager: Alertmanager is used to manage alerts from Prometheus. You can use it to receive alerts from multiple Prometheii. You can deduplicate, group, and route alerts to your integrations such as email, Slack, and PagerDuty. You can also configure Alertmanager to silence and inhibit specific alerts.
```
